languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -341,30 +341,129 @@ title="Easy editing stylesheet" ?>
 
 
  <category id='GRAMMAR' name="Gramática Geral" type="grammar">
-     <rule id="CONFUSION_OF_MAS_MAIS" name="confusion of mais and mas" default="temp_off">
+     <rule id="CONFUSION_OF_MAS_MAIS_COMMA" name="confusion of mais and mas after comma" default="temp_off">
         <antipattern>
+            <token>,</token>
             <token>mais</token>
             <token postag='AQ.+' postag_regexp='yes'/>
             <example>Stefani Joanne, mais conhecida como Lady Gaga, é uma cantora estadunidense.</example>
         </antipattern>
         <antipattern>
             <token regexp='yes'>quanto|quem</token>
-            <token skip="-1">mais</token>
+            <token skip="-1" postag='RG|AQ0CS0' postag_regexp='yes'/>
             <token>mais</token>
             <example>Quanto mais eu durmo, mais sono tenho.</example>
         </antipattern>
+        <antipattern>
+            <token>à</token>
+            <token>medida</token>
+            <token skip="-1">que</token>
+            <token>mais</token>
+            <example>À medida que o tempo passa, mais esperança tenho.</example>
+        </antipattern>
         <antipattern>
             <token skip="-1">mais</token>
             <token>,</token>
             <token>mais</token>
             <example>Este mercado tem mais ofertas, mais opções e mais economia.</example>
         </antipattern>
         <antipattern>
+            <token>mais</token>
+            <token>e</token>
+            <token>mais</token>
+            <example>Quando penso nos motivos, mais e mais compreendo.</example>
+        </antipattern>
+        <antipattern>
+            <token>,</token>
+            <token skip="-1">mais</token>
+            <token>e</token>
+            <token postag='N.+' postag_regexp='yes'/>
+            <example>Eram três quartos, mais sala de jantar e escritório.</example>
+        </antipattern>
+        <antipattern>
+            <token>menos</token>
+            <token postag='N.+' postag_regexp='yes'/>
+            <token>,</token>
+            <token>mais</token>
+            <token postag='N.+' postag_regexp='yes'/>
+            <example>Menos conversa, mais ação.</example>
+        </antipattern>
+        <antipattern>
+            <token>,</token>
+            <token>mais</token>
+            <token regexp='no' min='0'>de</token>
```

Comment:
I think `regexp="no"` is the default. You do not need to put that in.

Reply:
Thanks, I have fixed them all :)